test/cypress/integration/group-proposals.spec.js

```diff
@@ -185,7 +207,143 @@ describe('Proposals - Add members', () => {
+    cy.getByDT('openIncomeDetailModal').click()
+    cy.getByDT('dontNeedsIncomeRadio').click()
+    cy.getByDT('introIncomeOrPledge').should('contain', 'What\'s your monthly income?')
+    cy.getByDT('inputIncomeOrPledge').type(500)
+    cy.getByDT('badIncome').should('contain', 'It seems your income is not lower than the group\'s mincome.')
+    cy.getByDT('inputIncomeOrPledge').clear().type(100)
+    cy.getByDT('badIncome').should('not.be.visible')
+    cy.getByDT('submitIncome').click()
+    cy.getByDT('headerNeed').should('contain', 'You need $300')
+  })
+
+  it('It should display that no one is pledging money at the moment.', () => {
+    cy.get('.giving p').should('contain', 'No one needs monetary contributions at the moment.')
+    cy.get('.receiving p').should('contain', 'No one is pledging money at the moment.')
+  })
+
+  it('It should add non monetary contribution', () => {
+    cy.getByDT('addNonMonetaryContribution').last().click()
+    cy.getByDT('inputNonMonetaryContribution').type('Portuguese classes')
+
+    cy.getByDT('buttonAddNonMonetaryContribution').click()
+    cy.getByDT('givingList')
+      .get('li.is-editable')
+      .should('have.length', 1)
+      .should('contain', 'Portuguese classes')
+  })
+
+  it('It should remove non monetary contribution', () => {
+    cy.getByDT('buttonEditNonMonetaryContribution').click()
+    cy.getByDT('buttonRemoveNonMonetaryContribution').click()
+    cy.getByDT('givingList')
+      .get('li.is-editable')
+      .should('have.length', 0)
+  })
+
+  it('It should add the same non monetary contribution', () => {
+    cy.getByDT('addNonMonetaryContribution').click()
+    cy.getByDT('inputNonMonetaryContribution').type('Portuguese classes')
+
+    cy.getByDT('buttonAddNonMonetaryContribution').click()
+    cy.getByDT('givingList')
+      .get('li.is-editable')
+      .should('have.length', 1)
+      .should('contain', 'Portuguese classes')
+  })
+
+  it('It should edit the non monetary contribution', () => {
+    cy.getByDT('buttonEditNonMonetaryContribution').click()
+    cy.getByDT('inputNonMonetaryContribution').clear().type('French classes')
+    cy.getByDT('buttonSaveNonMonetaryContribution').click()
+
+    cy.getByDT('givingList')
+      .get('li.is-editable')
+      .should('have.length', 1)
+      .should('contain', 'French classes')
+  })
+
+  it('It should cancel the edit', () => {
+    cy.getByDT('buttonEditNonMonetaryContribution').click()
+    cy.getByDT('buttonCancelNonMonetaryContribution').click()
+    cy.getByDT('givingList')
+      .get('li.is-editable')
+      .should('have.length', 1)
+      .should('contain', 'French classes')
+  })
+
+  it('It should add more non monetary contribution', () => {
+    cy.getByDT('addNonMonetaryContribution').last().click()
+    cy.getByDT('inputNonMonetaryContribution').type('German classes')
+    cy.getByDT('buttonAddNonMonetaryContribution').click()
+
+    cy.getByDT('addNonMonetaryContribution').last().click()
```

Comment:
There's only one button addNonMonetaryContribution, so we don't need `.last()` here.

A tip: Usually when testing with cypress it's also a good practice to make sure the markup is correct (because a11y) for example, in this case, to make sure  addNonMonetaryContribution is a button, or givingList is a ul... `cy.getByDT` accepts as 2nd argument any selector type to match the data-test. For example:
- `cy.getByDT('givingList', 'ul')`
- `cy.getByDT('addNonMonetaryContribution', 'button')`

We don't use it all the time, but from time to time is okay just for sanity check :)